File: src/main/java/com/tibiawiki/process/RetrieveCorpses.java
package com.tibiawiki.process;

import com.tibiawiki.domain.RetrieveAnyService;
import com.tibiawiki.domain.enums.InfoboxTemplate;
import com.tibiawiki.domain.factories.JsonFactory;
import com.tibiawiki.domain.mediawiki.ArticleRepository;
import org.json.JSONObject;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import java.util.List;
import java.util.stream.Collectors;

@Component
public class RetrieveCorpses extends RetrieveAnyService {

    @Autowired
    public RetrieveCorpses(ArticleRepository articleRepository, JsonFactory jsonFactory) {
        super(articleRepository, jsonFactory);
    }

    public List<String> getCorpsesList() {
        final List<String> corpsesCategory = articleRepository.getPageNamesFromCategory(InfoboxTemplate.CORPSE.getCategoryName());
        final List<String> listsCategory = articleRepository.getPageNamesFromCategory(CATEGORY_LISTS);

        return corpsesCategory.stream()
                .filter(page -> !listsCategory.contains(page))
                .collect(Collectors.toList());
    }

    public List<JSONObject> getCorpsesJSON() {
        return getArticlesFromInfoboxTemplateAsJSON(getCorpsesList());
    }

    public JSONObject getCorpseJSON(String pageName) {
        return super.getArticleAsJSON(pageName);
    }
}

File: src/main/kotlin/com/tibiawiki/process/RetrieveCorpses.kt
package com.tibiawiki.process

import com.tibiawiki.domain.WikiJson
import com.tibiawiki.domain.enums.InfoboxTemplate
import com.tibiawiki.domain.factories.ArticleFactory
import com.tibiawiki.domain.factories.JsonFactory
import com.tibiawiki.domain.repositories.ArticleRepository
import org.springframework.stereotype.Component

@Component
class RetrieveCorpses(
    articleRepository: ArticleRepository,
    articleFactory: ArticleFactory,
    jsonFactory: JsonFactory
) : RetrieveAny(articleRepository, articleFactory, jsonFactory) {

    val corpsesList: List<String>
        get() {
            val category = articleRepository.getPageNamesFromCategory(InfoboxTemplate.CORPSE.categoryName)
            val listsCategory = articleRepository.getPageNamesFromCategory(CATEGORY_LISTS)
            return category.filter { page -> page !in listsCategory }
        }

    val corpsesJSON: List<WikiJson>
        get() = getArticlesFromInfoboxTemplateAsJSON(corpsesList)

    fun getCorpseJSON(pageName: String): WikiJson? {
        return getArticleAsJSON(pageName)
    }
}
